server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServer.java

```diff
@@ -210,9 +212,9 @@ protected ChannelUpstreamHandler createCoreHandler() {
         ImapChannelUpstreamHandler coreHandler;
         Encryption secure = getEncryption();
         if (secure != null && secure.isStartTLS()) {
-           coreHandler = new ImapChannelUpstreamHandler(hello, processor, encoder, compress, plainAuthDisallowed, secure, imapMetrics);
+           coreHandler = new ImapChannelUpstreamHandler(hello, processor, encoder, compress, plainAuthDisallowed, secure, plainAuthEnabled, imapMetrics);
```

Comment:
As said for Tung PR I would enjoy that we pass a configuration POJO around rather than individual parameters.

Reply:
I will solve this when rebasing Tung PR so I can reuse his code.

Reply:
resolved https://github.com/apache/james-project/pull/788